CombinatorialGames/Game/Specific/Domineering.lean

```diff
@@ -0,0 +1,155 @@
+/-
+Copyright (c) 2019 Kim Morrison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kim Morrison, Violeta Hernández Palacios
+-/
+import CombinatorialGames.Game.Concrete
+import Mathlib.Algebra.Group.Units.Equiv
+import Mathlib.Algebra.Order.Group.Nat
+
+/-!
+# Domineering as a combinatorial game.
+
+We define the game of Domineering, played on a chessboard of arbitrary shape
+(possibly even disconnected).
+Left moves by placing a domino vertically, while Right moves by placing a domino horizontally.
+
+This is only a fragment of a full development;
+in order to successfully analyse positions we would need some more theorems.
+Most importantly, we need a general statement that allows us to discard irrelevant moves.
+Specifically to domineering, we need the fact that
+disjoint parts of the chessboard give sums of games.
+-/
+
+namespace PGame
+
+open Function
+
+private theorem pred_fst_ne (m : ℤ × ℤ) : (m.1 - 1, m.2) ≠ m := by aesop
+private theorem pred_snd_ne (m : ℤ × ℤ) : (m.1, m.2 - 1) ≠ m := by aesop
+
+/-- A Domineering board is an arbitrary finite subset of `ℤ × ℤ`. -/
+def Domineering := Finset (ℤ × ℤ) deriving DecidableEq
```

Comment:
(I would like to add `Repr` here, though it doesn't seem `mathlib4` has Repr on `(ℤ × ℤ)`?)

Reply:
The problem isn't `ℤ × ℤ`, but `Finset`. There's an unsafe `Repr` instance in a file that wasn't previously imported - I've used it to construct the `Repr Domineering` instance.